### ISSUE TITLE
Bayesian Optimization search gets stuck in the same configuration

### ISSUE THREAD

**Issue by riccostamendes** [keras-team/keras-tuner#608]
I am using Bayesian Optimization to tune but the search got stuck in the same configuration, repeating it over and over again without trying new combinations. I am not using early stopping.

Search: Running Trial #99

Hyperparameter    |Value             |Best Value So Far 
num_layers        |17                |17                
units_0           |23                |23                
units_1           |39                |39                
units_2           |44                |44                
units_3           |25                |25                
units_4           |12                |12                
units_5           |25                |25                
units_6           |49                |49                
units_7           |15                |15                
units_8           |46                |46                
units_9           |36                |36                
units_10          |28                |28                
units_11          |12                |12                
units_12          |14                |14                
units_13          |31                |31                
units_14          |6                 |6                 
units_15          |35                |35                
units_16          |42                |42                
optimizer         |Adam              |Adam              
learning_rate     |0.001             |0.001             
lr_schedule       |False             |False             
mom_grad          |0.9               |0.9               
mom_grad_sq       |0.999             |0.999             
kernel_init       |RandomNormal      |RandomNormal      
bias_init         |Zeros             |Zeros             
dropout           |False             |False             
norm              |True              |True

**Comment by GearLord** [comment]
I have the same problem.
I tried with versions 1.0.3 and 1.0.4.
I couldn't try with version 1.1.0 because of another bug that isn't present in 1.0.3 and 1.0.4.
I will try version 1.1.1 to see if the problem has been resolved. I will let you know.

**Comment by GearLord** [comment]
I couldn't try v. 1.1.1 because I experience the same problem mentioned in issue #631 . 
I downgraded to v. 1.0.2 and now it works perfectly and it is really fast (tf version: 2.8.0).
On v. 1.0.3 and v. 1.0.4 it could take even 60 mins to generate a new combination of hyper parameters after 600 trials or so.
Now it does in hours what took days or even weeks before. 
I have no idea why, but now it works and it is really fast.

**Comment by jjohnarios** [comment]
I am facing the same issue. Bayesian Optimization keeps trying the same configuration and stops only when it reached the `max_trails` trials.

**Comment by thecorz** [comment]
Same problem here

**Comment by GearLord** [comment]
I have been using v. 1.0.2 for weeks and I can confirm that the bayesian optimization works fine on this version.
I noticed that the GPU isn't used when the bayesian optimizer is calculating a new combination (0% usage) of hyperparameters and all the work is done by the cpu that shows a significant increase in usage during the calculation of a new combination.
In the versions after 1.0.2 the load is shifted at least partially to the GPU (the usage is usually between 15% and 35% on those versions and the cpu usage is way lower than on v. 1.0.2.).
I hope this could be useful, unluckily I have not the skills required to even try to patch this.

**Comment by tyler-korenyi-both** [comment]
Same issue. Not a great fix, but I switched to RandomSearch and Hyperband instead.

**Comment by JohnJohnJohnJohn** [comment]
Same issue with 1.1.1, but RandomSearch works. My search space has 6 hp.Choice, 1 for set of str, 5 for set of int. Model is a simple convolutionary network

**Comment by VladimirVisnovsky** [comment]
Any updates? Still getting the same `stuck repeating` behaviour. From ~100 trials getting about 90 with the same parameters. Not using EarlyStopping.

**Comment by haifeng-jin** [comment]
Does that mean the Bayesian optimization has converged?

**Comment by VladimirVisnovsky** [comment]
Bayesian optimization finished it’s max_trials=100 iterations while exhausting ~15% of it’s search space.

So it seems bayesian converged but it doesn’t work at all. Why would it repeat the same parameter configuration 80/100 times. Also the parameters found by bayesian are far from optimal in comparison to Hyperband.

**Comment by haifeng-jin** [comment]
So we have two directions to solve this problem.
1) Just quit the search whenever it stucks at a local minimum.
2) Try to avoid that stuck by default setting. For example, dynamically decide the initial points or use exploration tricks to get out of the minimum when stucked.

Please may leave your thoughts on which way to go. : )

**Comment by Xargonus** [comment]
In my case, the Bayesian optimization repeated the same configuration 3 times in a row (at the time of writing), even though the configuration is far from perfect (0.78 maximizing metric vs 0.8 of the best trial).

I am for point 2, I want the optimizer to keep trying to find a better solution, not to quit prematurely.

Would it be possible to forbid Bayesian optimization from executing previously run configurations and pick the best configuration that hasn't been run before, or resort to random search instead if this is not possible?

**Comment by haifeng-jin** [comment]
@Xargonus Thanks!
That is possible.
What we can try to do is:
When BO keeps generate the same config, we just let it pick a random to expand exploration.

What do you think?

**Comment by Xargonus** [comment]
@haifeng-jin
I think that that is a good idea. Maybe a log mentioning that the optimizer is now doing a random trial would be helpful as well to get a better insight into what the optimizer is doing right now.

**Comment by muriloasouza** [comment]
Hello, any updates on this? I am still having this problem. Using keras-tuner v1.1.3.

**Comment by adrianrivadulla** [comment]
same problem with keras v1.1.3.

**Comment by SaiLalithPolawar** [comment]
I am also facing same issues. BO() taking same hyperparameter value combinations repeatedly. Not at all exploring the search space.

Any Update?

**Comment by lrvie** [comment]
Have you guys tried changing the `beta` parameter? It's supposedly responsible for the exploration-exploitation trade-off.
